### PR TITLE
cleaning and reducing repo

### PR DESCRIPTION
**Pull request by meghalithic**
cleaned out repository to limit it to the new scope

Here are questions and things left to do:
- is the OSC folder still needed?
- need to rerun the script to check that it works
  - scale and errors removed
  - renumber sampling.df
  - make sure heat maps are created
- where are the metadata files?
- how many images are removed after the scale bar and blobs are checked?
- should the repo be renamed to "minnow_segmented_traits"?

### CODE REVIEW

File: Scripts/Presence_Absence_Analysis.R [hunk 1]
```diff
@@ -27,49 +28,67 @@ sampling.df <- read.csv(file = sampling_path, header = TRUE)
...
+                       by.y = "original_file_name",
                        all.x = TRUE, all.y = FALSE)
 
-#### 8. sampling after segmentation ----
+## remove images that don't have a scale ----
+
+sampling.df$Selection_Criteria[10] <- "Has ruler scale"
+
+unique(presence.meta$unit) #None means no scale was detected or extracted
+errors <- presence.meta[presence.meta$unit == "None",] %>% drop_na(unit)
```
Comment (johnbradley) [inline Scripts/Presence_Absence_Analysis.R]: I had an error:
```
Error in drop_na(., unit) : could not find function "drop_na".
```
Here is one possible way to fix this issue:
```suggestion
errors <- presence.meta[presence.meta$unit == "None",] %>% tidyr::drop_na(unit)
```

Comment (johnbradley) [reply Scripts/Presence_Absence_Analysis.R]: After making the above change I encountered another error on the same line:
```
Error in `chr_as_locations()`:
! Can't subset columns that don't exist.
✖ Column `unit` doesn't exist.
Backtrace:
     ▆
  1. ├─presence.meta[presence.meta$unit == "None", ] %>% ...

```
I think the above error is due to expecting there to be a `unit` column in the `segmentation/Morphology/Presence/*_presence.json` files like there is for the other metadata files (eg. `segmentation/Metadata/INHS_FISH_100201.json`).

**Comment by johnbradley** [comment]
@megbalk See below for my thoughts on your questions:

> is the OSC folder still needed?

I don't think so. Users can just run this in their home directory. All data needed can be downloaded from our Dataverse instance.

> where are the metadata files?

Which metadata files are you asking about? After running the workflow the bgnn-formatted drexel_metadata files are in the subdirectory `segmentation/Metadata/` and the original drexel_metadata files are in the subdirectory `segmentation/DrexelMetadata/`.

> how many images are removed after the scale bar and blobs are checked?

Perhaps you could code something into the script to record this info.

> should the repo be renamed to "minnow_segmented_traits"?

That sounds good to me. The nice thing about renaming is GitHub will maintain the older repo names.

File: Scripts/Presence_Absence_Analysis.R (hunk 1 above)
Comment (johnbradley) [reply Scripts/Presence_Absence_Analysis.R]: I rebuilt the morphology-analysis container with @thibaulttabarin's fixes from https://github.com/hdr-bgnn/Morphology-analysis/pull/39. I think the `unit` column now needs to be  renamed `ruler_unit` now.

File: Scripts/Presence_Absence_Analysis.R [hunk 2]
```diff
@@ -27,49 +28,67 @@ sampling.df <- read.csv(file = sampling_path, header = TRUE)
...
+nrow(errors) #18 images
+
+write.csv(errors,
+          file = file.path(results, "df.missing.scale.csv"),
+          row.names = FALSE)
+
+presence.meta.scale <- presence.meta[!(presence.meta$base_name %in% errors$base_name),]
+
+write.csv(df.scale,
```
Comment (johnbradley) [inline Scripts/Presence_Absence_Analysis.R]: Should `df.scale` be `presence.meta.scale` ?

**Comment by meghalithic** [comment]
> i meant the metadata files that come from Tulane: Image_Metadata_v1_20211206_151152.csv and Image_Quality_Metadata_v1_20211206_151204.csv. I don't see them in the folder "Files" anymore

**Comment by johnbradley** [comment]
> i meant the metadata files that come from Tulane: Image_Metadata_v1_20211206_151152.csv and Image_Quality_Metadata_v1_20211206_151204.csv. I don't see them in the folder "Files" anymore

@megbalk These files are now stored in Dataverse: 
- https://datacommons.tdai.osu.edu/dataset.xhtml?persistentId=doi:10.5072/FK2/QOHJGD&version=DRAFT
- https://datacommons.tdai.osu.edu/dataset.xhtml?persistentId=doi:10.5072/FK2/ZIFDTJ&version=DRAFT

These input files are now downloaded as part of the first rules of the workflow.

Here is the config that describes the DOIs from Dataverse and the filenames:
https://github.com/hdr-bgnn/Minnow_Segmented_Traits/blob/78e665fceb1236efbd5491ae94393dcdb5b30d08/config/config.yaml#L4-L7

Here are the two rules that download the input files:
https://github.com/hdr-bgnn/Minnow_Segmented_Traits/blob/78e665fceb1236efbd5491ae94393dcdb5b30d08/workflow/Snakefile#L41-L57

**Comment by meghalithic** [comment]
i need to fix the readme to reflect where things are

**Comment by meghalithic** [comment]
@johnbradley i fixed the read me! I have place holders for when I rerun the code and will fix then.

**Comment by johnbradley** [comment]
The input CSV format and files have dramatically changed. The data is now spread over multiple CSV files and many column names have changed. There are changes here that rename the fields and read data from the new CSV files, but I don't think the changes are complete. 

I propose instead of reworking everything to fit the new format we could add a new first step that converts the new CSV input files into the old format. Then nothing else would need to change. Thoughts @megbalk?

**Comment by meghalithic** [comment]
@johnbradley yes, let's revert back. We can archive the images that are used in the study and for the segmentation model later.

**Comment by johnbradley** [comment]
To get started I'm going to create another branch based on the changes here. That way we can easily compare against these  changes to make sure we have everything we need. 

I think commit __modifying scripts__ https://github.com/hdr-bgnn/Minnow_Segmented_Traits/pull/30/commits/0f958035696abf0a7708ee33754095683701693a was the first that begin the transition to the new input format. I am going to create a new branch starting with commit https://github.com/hdr-bgnn/Minnow_Segmented_Traits/pull/30/commits/f9933879e5b504eb8b46fb4f7dfc319be85bad9b. You can browse the code at the point right now via https://github.com/hdr-bgnn/Minnow_Segmented_Traits/tree/f9933879e5b504eb8b46fb4f7dfc319be85bad9b.

**Comment by johnbradley** [comment]
I'm closing this PR since the work has been done via PR #33.

**Pull request closed without merging**